src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs

```diff
@@ -735,26 +735,12 @@ internal void ProcessNonSessionMessages(RemoteDataObject<PSObject> rcvdData)
 
         #region IDisposable
 
-        /// <summary>
-        /// Public method for dispose.
-        /// </summary>
-        public void Dispose()
-        {
-            Dispose(true);
-
-            GC.SuppressFinalize(this);
-        }
```

Comment:
The change is not related the PR purpose.

Reply:
The implementation is redundant in a sealed class.

Reply:
Specifically in a sealed class, `GC.SuppressFinalize` can be removed if the class has no unmanaged resources to dispose.

Reply:
https://rules.sonarsource.com/csharp/RSPEC-3234/

Reply:
Even if it's the right thing to do there is a risk. We do not have exhaustive tests for the remote code. 
I can merge `seal` only.

Reply:
Separated the change into https://github.com/PowerShell/PowerShell/pull/24705 so we can merge.

Reply:
> Even if it's the right thing to do there is a risk.

Similar changes were merged in https://github.com/PowerShell/PowerShell/pull/15802.